Model validation and testing

kennankole opened the pull request:
* Validated all the model fields 
* Written Unit tests using `RSpec` to test all the models and their methods

kennankole committed:
Created all model classes for all the entities in the ERD diagram
Set up all the associations between models

kennankole committed:
Added the following methods in the User and Post models.

  Users
    - created recent_posts method that returns the 3 most recent posts for a given user.
    - created update_post_counter method that updates the posts counter for a user
    - created recent_comments method that returns the 5 most recent comments for a given post.
  Posts
   - created update_comment_counter method that updates the comments counter for a post.
   - created update_likes_counter method that updates the likes counter for a post.

kennankole committed:
Adding Active Record validation for the
  - For the User model:
      name set not be blank.
      posts_counter set to be an integer greater than or equal to zero.

  - For the Post model:
      title set not be blank.
      title set not to exceed 250 characters.
      comments_counter set to be an integer greater than or equal to zero.
      likes_counter set to be an integer greater than or equal to zero.

kennankole committed:
Testing RSpec configuration

kennankole committed:
validates :comment_counter, presence: false, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
  validates :likes_counter, presence: false, numericality: { only_integer: true, greater_than_or_equal_to: 0 }

Adding Active Record validation for the
  - For the User model:
      name set not be blank.
      posts_counter set to be an integer greater than or equal to zero.

  - For the Post model:
      title set not be blank.
      title set not to exceed 250 characters.
      comments_counter set to be an integer greater than or equal to zero.
      likes_counter set to be an integer greater than or equal to zero.

Added Unit test for all the models and the respective methods

kennankole committed:
Modified rubocop.yml file to fix error
https://github.com/rubocop/rubocop/issues/9832
https://github.com/nedap/rubocop-config/commit/5112c907d8d60b85168090f5fa9e5f59aac6edd5

kennankole committed:
Updated the README.md file

kennankole committed:
Updating the README.md file according to PR comments

BenMukebo commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
- [ ] Great job so far 👏🏻 However let's make sure User Model specs have **Validation** tests to validate the presence of the user columns (name, post_counter,..) Example:

  ```ruby
  describe 'validations' do
    it { should validate_presence_of(:name) }
  end
  ```

BenMukebo reviewed (changes requested):
### Hi @kennankole,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- Good commit messages ✔️ 
- User model has validations. ✔️ 
- Post model has validations. ✔️ 
- All the custom methods have a corresponding unit test. ✔️ 
- All tests are passing🥇 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

kennankole committed:
Making changes according to PR comments
https://github.com/kennankole/my-blog-app/pull/3#:~:text=No%20linter%20errors-,Required%20Changes,-Check%20the%20comments

kennankole committed:
Merge branch 'processing-data' into model-validation-and-testing

kennankole committed:
Merge data-processing branch

kennankole committed:
Refactored unit tests

codepantha reviewed (approved):
# STATUS: APPROVED 🟢 ✔️ 

## Hi @kennankole 👋, 

Well done for implementing the required changes ♻️ 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media4.giphy.com/media/5oGIdt1xapQ76/giphy.gif"/>

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
